### DISCUSSION TITLE
Objective function is being calculated multiple times

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#387] [General]
Hi, 
I am running an airfoil shape optimization case. After calculating sensitivities, it start running CFD calculation (i.e Objective function calculation). I have set 5000 iterations and mesh size is 20,000 cells with C type domain. 
It is running multiple CFD runs. I am attaching the log file with this post.

Please guide,

Thanks
[Objective Function Error.txt](https://github.com/mdolab/dafoam/files/10399766/Objective.Function.Error.txt)

**Comment by friedenhe** [comment]
It is doing line search

**Reply by usmanshahid90** [reply]
what's that? is it steepest descent?? can you please elaborate?

**Reply by friedenhe** [reply]
Read https://mdobook.github.io/

**Reply by uttamcadambi07** [reply]
To put it simply - line search (or specifically backtracking line search) progressively decreases the value of the multiplicative constant that will be used to determine the extent of progress along a particular line. It usually starts with a very high value (in this case, say for example 1) and progressively decreases using this equation (2^{-n}) where n is the nth successive decrease in the parameter. To give you an example, if you look at a parabolic function which has a minima and you start at any one point, if your step size in the direction of the steepest descent is too large, you may miss the minima. So you start with a step size of 1, check the value of the function, if it is less than a threshold, then you move to the next step of the iteration. But if it is not less than that threshold, you reduce the step size by say half (0.5) and then do the check again.

Backtracking line search is very good when your functions are highly non-linear and non-convex where it could take up many iterations to converge to the local minimum. 

Hope this was a good quick explanation. For more, you can read the book.
